packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -141,17 +153,9 @@ export abstract class BaseServiceV2<
    * @param params.port Port for the app server. Defaults to 7300.
    * @param params.hostname Hostname for the app server. Defaults to 0.0.0.0.
    */
-  constructor(params: {
-    name: string
-    version: string
-    optionsSpec: OptionsSpec<TOptions>
-    metricsSpec: MetricsSpec<TMetrics>
-    options?: Partial<TOptions>
-    loop?: boolean
-    loopIntervalMs?: number
-    port?: number
-    hostname?: string
-  }) {
+  constructor(
+    private readonly params: BaseServiceV2Params<TOptions, TMetrics>
```

Comment:
I think it's ok to get rid of `private readonly` since we're parsing the params how we want them